### PR TITLE
[Overflow-25] Answer Component

### PR DESCRIPTION
**Pull request by presyos**
## Backlog Link
https://framgiaph.backlog.com/view/SUN_OVERFLOW-25
## Commands
cd frontend
npm i
## Pre-conditions
none
## Expected Output
Answer Component with  RichText Editor and a button for submitting an answer
## Notes
non
## Screenshots
image
image

### CODE REVIEW

File: frontend/components/atoms/Button/index.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    <button
+      type="submit"
+      className="items-center px-5 py-2.5 text-sm font-medium text-center text-red-700 rounded-lg border-2 border-red-500 focus:ring-1 focus:ring-red-600 hover:bg-rose-200 "
+    >
+      {propname}
+    </button>
+  );
+}
+
+export default Button;
```
Comment (kenorenzsun) [inline frontend/components/atoms/Button/index.tsx:15]: Add EOF

File: frontend/components/molecules/RichTextEditor/index.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  return (
+    <ReactQuill
+      className="border-2 border"
+      modules={modules}
+      formats={formats}
+    />
+  );
+}
+
+export default RichTextEditor;
```
Comment (kenorenzsun) [inline frontend/components/molecules/RichTextEditor/index.tsx:57]:
```suggestion
export default RichTextEditor;
```

add EOF

File: frontend/components/organisms/AnswerComponent/index.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        <RichTextEditor />
+        <div className="mt-2 flex flex-row-reverse">
+          <Button propname="Submit Answer" />
+        </div>
+      </div>
+    </div>
+  );
+}
+
+export default AnswerComponent;
```
Comment (kenorenzsun) [inline frontend/components/organisms/AnswerComponent/index.tsx:19]:
```suggestion
export default AnswerComponent;

```

File: frontend/components/atoms/Button/index.tsx (hunk 1 above)
Comment (kenorenzsun) [reply frontend/components/atoms/Button/index.tsx:15]: Is there only one type of button ? If there is multiple types of buttons (ex. primary, error/danger, success) please make it dynamic.

File: frontend/styles/globals.css [hunk 4]
```diff
@@ -1,107 +1,3 @@
...
-}
-
-@media (prefers-color-scheme: dark) {
-  html {
-    color-scheme: dark;
-  }
-}
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
```
Comment (kenorenzsun) [inline frontend/styles/globals.css:3]:
```suggestion
@tailwind utilities;

```

File: frontend/tailwind.config.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+module.exports = {
+  content: [
+    "./pages/**/*.{js,ts,jsx,tsx}",
+    "./components/**/*.{js,ts,jsx,tsx}",
+  ],
+  theme: {
+    extend: {},
+  },
+  plugins: [],
+};
```
Comment (kenorenzsun) [inline frontend/tailwind.config.js:11]:
```suggestion
};

```

File: frontend/components/atoms/Button/index.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import * as React from "react";
+
+function Button(props) {
```
Comment (nicoleamber) [inline frontend/components/atoms/Button/index.tsx:3]: Please use arrow functions. Also, destructure the props ahead. Apply to all.
```suggestion
const Button = ({ propsname }) => {

```

File: frontend/components/atoms/Button/index.tsx [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+        >
+          {title}
+        </button>
+      );
+      break;
+    default:
+      return;
+      break;
+  }
+};
```
Comment (kenorenzsun) [inline frontend/components/atoms/Button/index.tsx:19]:
```suggestion
const Button = ({ usage= "primary", type = "submit", additionalClass, isDisabled = false, children, onClick }) => {
  const getClass = () => {
    if(usage === "primary"){
      return `bg-blue-600 py-2 hover:bg-blue-800`
    }
  }
  
  return (
    button 
            onClick={onClick}
            disabled={isDisabled}
            type={type}
            className={`text-white font-bold py-2 rounded-lg ${getClass()} ${additionalClass}`}
    >
            {children}
    </button>
  )
 
}

```
You have to identify the common classes that are used for the buttons and put them in the className. Then for the unique classes for usage such as Primary, Secondary, Success buttons you can put then in the getClass function

**Review by nicoleamber — APPROVED** [review]

**Pull request merged**